### ISSUE TITLE
Consider (again) using rugged

### ISSUE THREAD

**Issue by matt-harvey** [matt-harvey/git_curate#12]
I'm thinking `git_curate` should move to using `https://github.com/libgit2/rugged` instead of parsing the output of the `git` CLI. The latter approach is starting to feel unmaintainably brittle, especially in light of Issues #10 and #11.

Using `rugged` would probably make `git_curate` faster as well.

I previously decided against `rugged` (see #6), out of a desire to make the installation process for `git_curate` as simple as possible; but I'm rethinking that now in light. I think the installation complexity consideration is fairly easy to mitigate (with clear instructions, providing a Homebrew, Snap etc. packages), but brittleness and sluggishness and much harder to live with over the long term.

**Comment by matt-harvey** [comment]
`rugged` has now been added as a dependency, and is used for some operations. The `v1.2.0.beta3` release contains these changes.

Certain operations still call out to the `git` CLI, however, and probably always will. For example, when I attempted to use `rugged` to obtain the shortened hash for a given commit in a manner that reliably matches `git`'s ordinary output of same, it succeeded for some repositories, but failed for others (e.g. the Linux kernel repository, which contains a very large number of commits). In contrast, using `git` CLI operations to obtain shortened commit hashes works reliably and performantly even on all repositories tested. So, for the forseeable future `git_curate` will continue using `git` CLI calls for obtaining shortened hashes, among other things.

There is no real performance improvement in `v1.2.0.beta3`, unfortunately, but my hope is that the use of `rugged` will make `git_curate` more robust, by avoiding the need to parse CLI output for certain operations, given this output seems sometimes to vary between platforms.
